### DISCUSSION TITLE
Error on Windows Install/ Command Prompt

### DISCUSSION THREAD

**Discussion by Sierevello** [fra589/cn5X#41] [General]
I have tried to install on Windows using the Wiki instructions with no luck. When I double click on the cn5X.py file I just get a quick blip on the screen and then nothing. I followed the advice to locate the error by trying to run the cn5X.py file through the admin command prompt and that did not work either. However, I was able to find the error that is keeping the program from running. I have attached it to this message. Thank you for any help on getting this working and THANK YOU for all of your hard work on this project.

-Steve
![image_2023-03-21_131302947](https://user-images.githubusercontent.com/67448595/226688730-4e328b1f-53a8-4753-b03d-5b8e9e8fff67.png)

**Comment by fra589** [comment]
@Sierevello, Hi Steve,  
  
You don't have the up to date version of cn5X++...
Your version seams to be 0.8.4. The error message you get corresponds to a known bug (#36) which was fixed a little over a year ago.
   
You must download the latest version (v0.8.9) which is deemed to work with Windows.  
  
@++;  
Gauthier.

**Reply by Sierevello** [reply]
When I go to Github it shows only 0.8.4 as the "latest". Am I looking in the wrong place? I have attached a screenshot of what I see. Thank you for your quick response!

-Steve

**Comment by Sierevello** [comment]
![cn5x](https://user-images.githubusercontent.com/67448595/226905637-e6a2b075-e4ef-414d-9c91-531b0000a174.PNG)

**Comment by fra589** [comment]
hi @Sierevello,  
  
Oh ! OOPS ! This was published by mistake... I will delete this past version.
You must download the latest version by clicking on "<> Code'", then "Download ZIP"  on the main Github cn5X page :
![image](https://user-images.githubusercontent.com/19529154/226932671-67ba9b6c-f2b8-4996-922a-e41cbc79d4b7.png)


@++;
Gauthier.

**Comment by Sierevello** [comment]
Thank You!!!

Ha!!! It worked! However, I was not able to open the program by double clicking on cn5X.py.

I had to open it from the command prompt by using python cn5X.py I was able to open from normal command prompt and also administrator command prompt. Can you tell me how to open it by double clicking on the cn5X.py file? When I double click the file it just gives me a quick "blip" on the screen and goes away.

Also, there was a warning in the cn5X++ log when I connected to the Mega, I have attached a screen shot. 

I was able to jog my x-axis that I had set up, but I was not able to hold down on the x-axis button and continually jog until released. It would only jog the pre-determined distance I had checked in "Jog Increment". Is there a way to do that?

Thank You for all of your help!!! What an amazing piece of work!

-Steve
![cn5x error](https://user-images.githubusercontent.com/67448595/226977039-b9f0e875-6d43-4060-ab8e-b47d67f3a3b0.PNG)

**Comment by fra589** [comment]
Hi @Sierevello,  
  
The .py file association to be able to run Python script by double clicking on it is a Python install option.  
If you don't select it when you installed Python, you can add this file association like this:
  
On Windows its very easy to run on double click:  
1- Just Right Click the script file and go to properties.  
2- Select the option 'Opens with:' in General tab, and select the python from list, if its not available then browse to the installation directory of python and select the python.exe from there.  
cf. Google: https://www.google.com/search?client=firefox-b-d&q=run+python+script+on+double+click  
  
The warning that appears when connecting Grbl happens quite often with some Arduino boards.
I don't know exactly why this happens... Maybe a little too much delay when initializing the serial interface?
Or else, during a corruption of the flash memory of the Arduino card. In this case, we can reset the flash with the Grbl command: `$RST=*`  
  
To continuously jog when hold down jog buttons until they are released, you need to select the "+0.00" increment jog.  
  
@++;  
Gauthier.

**Comment by Sierevello** [comment]
Good Morning,

I tried that and still get the same result. I am also sure I clicked the appropriate boxes when I installed Python because I read the discussion before I installed and saw the issues you talked about. Any idea what to try next?

Thank You!!!

-Steve
![Python](https://user-images.githubusercontent.com/67448595/227220799-978bbbb2-358f-4242-8e13-921f68c2aacd.PNG)

**Comment by Sierevello** [comment]
I just tried uninstalling Python and all dependencies and reinstalling and still no luck. I can run from the console only.

Thanks!

-Steve

**Comment by Sierevello** [comment]
I have also checked PATH and Python is there. Not sure what is going on.
![Path](https://user-images.githubusercontent.com/67448595/227343474-6d2183b0-14bd-465a-bc7a-5f2dc58e1aad.PNG)

**Comment by Sierevello** [comment]
I wrote a batch file to open the cn5X.py and when I ran it as an administrator it gave me an error, It can't seem to locate the directory? I have attached the error it gave me in the command prompt.

Thanks,
Steve
![image_2023-03-23_181727177](https://user-images.githubusercontent.com/67448595/227377111-9a5511eb-8015-4bba-beb8-cdd25ae7321b.png)

**Comment by fra589** [comment]
Hi @Sierevello,  
  
\<Joke\>Grrr, F...ing Windows! You should use Linux...\</Joke\>

In my old Windows10 box, cn5X have always worked with a double click on the .py file. To understand your problem, I uninstalled python and redo the full install.
After this, cn5X stopped working with the double click on its file! 😞😭 

Finally, I found the problem why double clicking on cn5X.py don't work:  
In almost all Linux distributions, Python is installed by default, but a majority still have python2 and python3 installed together. And python3 is not necessarily the default version!
So, in the cn5X.py file, the first line is `#! /usr/bin/env python3` to force the use of the Python version 3.

After many tests, I found that unfortunately, Windows does'nt reconize the command <python3>, but only the <python> command!

The workaround to make Windows finding python3 is to go to the install directory of Python, then, copy the python.exe file to the name python3.exe.  
With this, double clicking on the cn5X.py file worked on my box! 😃  

@++;
Gauthier.  
  
PS As Python2 is end of life (https://wiki.debian.org/Python/FAQ#Python_2_support), the next cn5X release will probably change the way to find the Python executable in the first line of the script and the solution workaround will probably no longer be necessary...

**Comment by Sierevello** [comment]
Good Afternoon,

I am a little confused on which name to use. I tried to replace python3 with python 3.11.2 and it did not work. I have attached screenshots of the folder where python is and it just lists the name as python.exe. I also attached pics with the folder details. Thank you again for the help!

-Steve
![File](https://user-images.githubusercontent.com/67448595/227592305-c38386a4-cd3b-41ab-a278-6d939a414ee7.PNG)
![Details](https://user-images.githubusercontent.com/67448595/227592326-f8378c78-83dd-41f8-9c17-b2bac09801bb.PNG)
![General](https://user-images.githubusercontent.com/67448595/227592348-d733d98f-7de8-423c-bab2-d0af9e5e9cc0.PNG)

**Comment by fra589** [comment]
Sorry or my bad English spoke, but what's you don't understand in "**_copy the python.exe file to the name python3.exe_**" ?

=> Open a command prompt (cmd.exe) then type on your keyboard (the thing with lots of keys usually placed between your chair and the computer screen):
- cd C:\Users\Metalmovers\AppData\Local\Programs\Python\Python311
- copy python.exe python3.exe

Then make a new test by clicking on cn5X.py file in the Windows file manager...

**Comment by Sierevello** [comment]
That did it! Thank you for all of your help and also for creating such a great program!!!

-Steve
